modules/core/src/test/java/org/locationtech/jts/index/kdtree/KdTreeTest.java

```diff
@@ -103,6 +107,111 @@ public void testSizeDepth() {
     assertTrue( depth <= size );
   }
   
+  public void testNearestNeighbor() {
+      int n = 1000;
+      int queries = 500;
+      KdTree tree = new KdTree();
+      Random rand = new Random(1337);
+
+      for (int i = 0; i < n; i++) {
+          double x = rand.nextDouble();
+          double y = rand.nextDouble();
+          tree.insert(new Coordinate(x, y));
+      }
+
+      for (int i = 0; i < queries; i++) {
+          double queryX = rand.nextDouble();
+          double queryY = rand.nextDouble();
+          Coordinate query = new Coordinate(queryX, queryY);
+
+          KdNode nearestNode = tree.nearestNeighbor(query);
+
+          Coordinate bruteForceNearest = bruteForceNearestNeighbor(tree, query);
+          
+          assertEquals(nearestNode.getCoordinate(), bruteForceNearest);
+      }
+  }
+  
+  public void testNearestNeighbors() {
+      int n = 2500;
+      int numTrials = 50;
+      Random rand = new Random(0);
+
+      for (int trial = 0; trial < numTrials; trial++) {
+          KdTree tree = new KdTree();
+
+          for (int i = 0; i < n; i++) {
+              double x = rand.nextDouble();
+              double y = rand.nextDouble();
+              tree.insert(new Coordinate(x, y));
+          }
+
+          Coordinate query = new Coordinate(rand.nextDouble(), rand.nextDouble());
+          int k = rand.nextInt(n/10);
+          
+          List<KdNode> nearestNodes = tree.nearestNeighbors(query, k);
+
+          List<Coordinate> bruteForceNearest = bruteForceNearestNeighbors(tree, query, k);
+
+          assertEquals(k, nearestNodes.size());
+          for (int i = 0; i < k; i++) {
+              assertEquals(bruteForceNearest.get(i), nearestNodes.get(i).getCoordinate());
+          }
+      }
+  }
+  
+  public void testNearestNeighborsPerformance() {
```

Comment:
i keep performance tests out of unit test classes, since they are usually time-consuming.  They go in the package [`test.jts.perf`](https://github.com/locationtech/jts/tree/master/modules/core/src/test/java/test/jts/perf).

Reply:
This one's only about 1s. Should still be moved?

Reply:
Yes, please.